nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection-configuration.js

```diff
@@ -102,7 +102,7 @@
                 initializeSourceProcessor(source).done(function (processor) {
                     if (!nfCommon.isEmpty(processor.relationships)) {
                         // populate the available connections
-                        $.each(processor.relationships, function (i, relationship) {
+                        $.each(processor.relationships.sort(), function (i, relationship) {
```

Comment:
Same as within the connection, I believe the sort is happening server side [1] and that we should fix the merging to ensure the REST API is consistent regardless of standalone vs clustered.

[1] https://github.com/apache/nifi/blob/161e4b57632b2d5454390871980288eb66df2502/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/dto/DtoFactory.java#L2870

Reply:
Did you actually encounter the processor relationships not being sorted correctly? I'm actually not sure if my previous comment is correct. The DTO there should be using a `List` which should preserve the order correctly.

Reply:
@mcgilman, I observed the out-of-order of Relationships only in a cluster environment. I could not reproduce in a stand-alone. The ordering is not the list of processor properties (e.g. RouteOnAttribute with several user-defined relationships.) Rather, it is the "For Relationships" list on the Connection configuration. Moreover, the list appears correctly sorted when creating a new relationship. However, when returning to the configuration of the same Connection, the "For Relationships" is now out of order. This observation also points to a problem with the Cluster merging logic since when creating a new Connection, the Cluster is not queried (I don't believe), but it is when viewing the configuration of a Connection.
I think you are correct in this  having to do with it being a Set (not ordered) in the DTO. Specifically, I noted ConnectionDTO.getAvailableRelationships method returns a Set. Perhaps this should be a List instead?

Reply:
@markobean Right so there should be no change needed on the `ProcessorDTO.setRelationships` side since order should be preserved when the object is unmarshalled.

On the connection side, unfortunately, we cannot simply change the type in the DTO since we regard that object model as part of our REST API public interface. We cannot break those APIs until our next major release.

However, as I alluded to earlier, we can enforce the order when merging the responses similar to when we serialize the DTO in the linked source above. That change would need to be added when merging responses that contain any `ConnectionDTO`. That change would belong in `ConnectionEntityMerger` [1]. 

Is that something that can be updated in this PR? I understand it's a bit outside the scope of this proposed PR so I'm happy to have this PR closed the original JIRA updated to reflect these findings.

[1] https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/manager/ConnectionEntityMerger.java

Reply:
I agree with closing this PR, and re-working from ConnectionEntityMerger rather than covering it up in javascript.